src/bundle/Resources/public/ts/shared/BaseBadge.ts

```diff
@@ -0,0 +1,31 @@
+import Base from './Base';
+
+const MAX_BADGE_VALUE = 99;
+
+export default abstract class BaseBadge extends Base {
+    private value = '';
```

Comment:
considering we currently accept only numbers (as we dropped flexibility as currently not needed) we can assume it's number (integer?) instead of string. And then you'll be able to drop _parseValue as .toString() as well :)